mlflow/transformers.py

```diff
@@ -0,0 +1,990 @@
+
+    with path.joinpath(_CONDA_ENV_FILE_NAME).open("w") as f:
+        yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
+
+    if pip_constraints:
+        write_to(str(path.joinpath(_CONSTRAINTS_FILE_NAME)), "\n".join(pip_constraints))
+
+    write_to(str(path.joinpath(_REQUIREMENTS_FILE_NAME)), "\n".join(pip_requirements))
+
+    _PythonEnv.current().to_yaml(str(path.joinpath(_PYTHON_ENV_FILE_NAME)))
+
+
+@experimental
+@docstring_version_compatibility_warning(integration_name=FLAVOR_NAME)
+@format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name=FLAVOR_NAME))
+def log_model(
+    transformers_model,
+    artifact_path: str,
+    processor=None,
+    task: Optional[str] = None,
+    model_card=None,
+    code_paths: Optional[List[str]] = None,
+    registered_model_name: str = None,
+    signature: Optional[ModelSignature] = None,
+    input_example: Optional[ModelInputExample] = None,
+    await_registration_for=DEFAULT_AWAIT_MAX_SLEEP_SECONDS,
+    pip_requirements: Optional[Union[List[str], str]] = None,
+    extra_pip_requirements: Optional[Union[List[str], str]] = None,
+    conda_env=None,
+    metadata: Dict[str, Any] = None,
+    **kwargs,
+):
+    """
+    Log a ``transformers`` object as an MLflow artifact for the current run.
+
+    :param transformers_model:
+        A trained transformers `Pipeline` or a dictionary that maps required components of a
+        pipeline to the named keys of ["model", "image_processor", "tokenizer",
+        "feature_extractor"]. The `model` key in the dictionary must map to a value that inherits
+        from `PreTrainedModel`, `TFPreTrainedModel`, or `FlaxPreTrainedModel`.
+        All other component entries in the dictionary must support the defined task type that is
+        associated with the base model type configuration.
+
+        An example of supplying component-level parts of a transformers model is shown below:
+
+        .. code-block:: python
+
+          from transformers import MobileBertForQuestionAnswering, AutoTokenizer
+
+          architecture = "csarron/mobilebert-uncased-squad-v2"
+          tokenizer = AutoTokenizer.from_pretrained(architecture)
+          model = MobileBertForQuestionAnswering.from_pretrained(architecture)
+
+          with mlflow.start_run():
+              components = {
+                  "model": model,
+                  "tokenizer": tokenizer,
+              }
+              mlflow.transformers.log_model(
+                  transformers_model=components,
+                  artifact_path="my_model",
+              )
+
+        An example of submitting a `Pipeline` from a default pipeline instantiation:
+
+        .. code-block:: python
+
+          from transformers import pipeline
+
+          pipeline = "csarron/mobilebert-uncased-squad-v2"
+
+          with mlflow.start_run():
+              mlflow.transformers.log_model(
+                  transformers_model=pipeline,
+                  artifact_path="my_pipeline",
+              )
```

Comment:
This seems to fail (https://github.com/mlflow/mlflow/pull/8086/files#r1152507646)